test/unit/org/apache/cassandra/index/sai/cql/StaticColumnIndexTest.java

```diff
@@ -49,7 +49,26 @@ public void staticIndexAndNonStaticIndex() throws Throwable
         execute("INSERT INTO %s(pk, ck,       val2) VALUES(?, ?,    ?)", 1, 2,     2000);
         execute("INSERT INTO %s(pk, ck, val1, val2) VALUES(?, ?, ?, ?)", 2, 1, 40, 2000);
 
-        beforeAndAfterFlush(() -> assertRows(execute("SELECT pk, ck, val1, val2 FROM %s WHERE val1 = 20 AND val2 = 2000 ALLOW FILTERING"),
+        beforeAndAfterFlush(() -> assertRows(execute("SELECT pk, ck, val1, val2 FROM %s WHERE val1 = 20 AND val2 = 2000"),
                                              row(1, 2, 20, 2000)));
     }
+
+    @Test
+    public void staticAndNonStaticRangeIntersection() throws Throwable
```

Comment:
This is a concrete reproduction of the bug Harry found (see https://github.com/apache/cassandra/pull/3044)